crates/ruff_linter/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -800,7 +803,211 @@ pub(crate) fn fix_unnecessary_map(
         }
     }
 
-    Ok(Edit::range_replacement(content, expr.range()))
+    Ok(Edit::range_replacement(content, range))
+}
+
+/// (C417) Return a fix that converts:
+///
+/// ```python
+/// # Before                               # After
+/// map(lambda x: x * 2, a)                # (x * 2 for x in a)
+/// map(lambda x, y: (x, y), a, b)         # ((x, y) for x in zip(a, b))
+/// list(map(lambda x: x * 2, a))          # [x * 2 for x in a]
+/// list(map(lambda x, y: (x, y), a, b))   # [(x, y) for x in zip(a, b)]
+/// set(map(lambda x: x * 2, a))           # {x * 2 for x in a}
+/// set(map(lambda x, y: (x, y), a, b))    # {(x, y) for x in zip(a, b)}
+/// dict(map(lambda x, y: (x, y), a, b))   # {x: y for x, y in zip(a, b)}
+/// ```
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn fix_unnecessary_map_preview(
+    range: TextRange,
+    parent: Option<&Expr>,
+    object_type: ObjectType,
+    lambda: &ast::ExprLambda,
+    parameters: &Parameters,
+    body: &Expr,
+    iterables: &[Expr],
+    checker: &Checker,
+) -> Result<Edit> {
+    let generator = checker.generator();
+    let semantic = checker.semantic();
+
+    let loop_iter = if let [iterable] = iterables {
+        iterable.clone()
+    } else if semantic.has_builtin_binding("zip") {
+        ast::ExprCall {
+            func: Expr::Name(ast::ExprName {
+                id: ast::name::Name::new("zip"),
+                ctx: ExprContext::Load,
+                range: TextRange::default(),
+            })
+            .into(),
+            arguments: ast::Arguments {
+                args: iterables.into(),
+                keywords: [].into(),
+                range: TextRange::default(),
+            },
+            range: TextRange::default(),
+        }
+        .into()
+    } else {
+        bail!("Cannot create fix: `zip` is overshadowed");
+    };
```

Comment:
theoretically we could try to import `builtins` and then use the fully qualified `builtins.zip` here if `zip` is shadowed. I don't think it's something for this PR, though, and it may not be worth the complexity to ever do it. Just a thought.

Also, a nitpick: I'd probably just say "shadowed" rather than "overshadowed"

```suggestion
    } else {
        bail!("Cannot create fix: `zip` is shadowed locally");
    };
```